torchao/core/config.py

```diff
@@ -253,10 +244,11 @@ def config_from_dict(data: Dict[str, Any]) -> AOBaseConfig:
             f"Failed to find class {type_path} in any of the allowed modules: {allowed_modules_str}"
         )
 
-    # Check version - require exact match
```

Comment:
I feel like in general versions will error right? like shouldnt we have this path but just let instances figure out if they should error or not at deserialization

Reply:
they can still error, see quant_api.py, they can error when `config.version` has a deprecated version number

but erroring out here would mean older version are not supported at all